CPAC/pipeline/nipype_pipeline_engine/monkeypatch.py

```diff
@@ -0,0 +1,65 @@
+def patch_base_interface():
+    """
+    Monkey-patch nipype.interfaces.base.core.BaseInterface.run().
+
+    This is a temporary workaround to prevent nipype from emitting
+    .proc-* files in the current working directory.
+    """
+    from nipype.interfaces.base.core import (
+        BaseInterface,
+        config,
+        indirectory,
+        InterfaceResult,
+        os,
+        RuntimeContext,
+        str2bool,
+        write_provenance,
+    )
+
+    class PatchedBaseInterface(BaseInterface):
+        def run(self, cwd=None, ignore_exception=None, **inputs):
+            with indirectory(cwd or os.getcwd()):  # This is the only line that changed
+                rtc = RuntimeContext(
+                    resource_monitor=config.resource_monitor and self.resource_monitor,
+                    ignore_exception=ignore_exception
+                    if ignore_exception is not None
+                    else self.ignore_exception,
+                )
+
+            with indirectory(cwd or os.getcwd()):
+                self.inputs.trait_set(**inputs)
```

Comment:
Any reason not to combine these two `with` blocks?

```suggestion
            with indirectory(cwd or os.getcwd()):
                # moved rtc init into context
                rtc = RuntimeContext(
                    resource_monitor=config.resource_monitor and self.resource_monitor,
                    ignore_exception=ignore_exception
                    if ignore_exception is not None
                    else self.ignore_exception,
                )
                self.inputs.trait_set(**inputs)
```

Reply:
Yeah this way I could comment `# this is the only line that changed` and future diffs would be marginally simpler. I also did not want to think about if some overloaded constructor might have the ability to change the working dir and it needs to be reset again (probably not. Either way I have no hard feelings